xapian-core/api/valuerangeproc.cc

```diff
@@ -81,11 +81,11 @@ decode_xxy(const string & s, int & x1, int &x2, int &y)
     if (s.size() - j > 4 + 1) return false;
     if (s.find_first_not_of("0123456789", j + 1) != string::npos)
 	return false;
-    x1 = atoi(s.c_str());
+    x1 = stoi(s);
```

Comment:
`std::stoi()` is a C++11 library function, but it reports errors by throwing an exception instead of via its return value like the C `strtol()` and related functions do.

C++ exception handling has a significant overhead (implementations are optimised to make the case where an exception is not thrown and caught faster at the expense of making throwing and catching an exception slower), so we want to avoid `std::stoi()` and related functions at least in cases where not getting a number is a normal case.  It also seems that some implementations don't yet have `stoi()` (see the appveyor CI failures you're getting).

Aside from the current lack of portability, it would be OK to use `stoi()` in cases where not getting a number is genuinely exceptional, though you really need to ensure that the exception is usefully caught and reported.